src/markov/sean.clj

```diff
@@ -0,0 +1,98 @@
+(ns markov.sean
+  (:gen-class))
+
+;; Generating a markov dictionary from a string of text
+;; e.g. { :first-words ["The" "An"] :last-words ["dog"] :dict { "The" ["quick" "large"] "fox" ["jumped"] ... } }
+
+(defn zip-pairs
+  "From a list, generates a list of sequential pairs ([1 2 3] becomes [[1 2] [2 3]])"
+  [list]
+  (map vector list (next list)))
+
+(defn build-dict
+  "Builds the graph of words in a markov dictionary, given an ordered list of words"
+  [words]
+  (reduce (fn [dict, current-pair]
+            (let [current-word (current-pair 0)
+                  next-word (current-pair 1)
+                  words-after-current (get dict current-word)]
+              (if words-after-current
+                (update dict current-word conj next-word) 
+                (merge dict {current-word [next-word]}))))
+          {} (zip-pairs words)))
+
+(defn split-into-words
+  [text]
+  (clojure.string/split text #" "))
+
+(defn get-first-words
+  "From a list of sentences, returns a list of the first words"
+  [sentences]
+  (map (fn [sentence]
+         (first (split-into-words sentence))) sentences))
```

Comment:
I'm not sure which I'd go with but I think this could also be written as:

```clojure
(for [[first-word & _] sentences]
  first-word)
```
It would also work without the `& _` but I think that makes it a bit more explicit to the reader that you expect there to be trailing words and you're ignoring them.